rhel/vex/fetcher.go

```diff
@@ -62,48 +74,46 @@ func (u *Updater) Fetch(ctx context.Context, hint driver.Fingerprint) (io.ReadCl
 		}
 	}()
 
+	var compressedURL *url.URL
 	// Is this the first run or has the updater changed since the last run?
-	if fp.changesEtag == "" || fp.version != updaterVersion {
+	processArchive := fp.changesEtag == "" || fp.version != updaterVersion
+	if processArchive {
 		// We need to go after the full corpus of vulnerabilities
-		// First we target the archive_latest.txt file
-		latestURI, err := u.url.Parse(latestFile)
+		// First we target the archive_latest.txt file.
+		var err error
+		compressedURL, err = u.getCompressedFileURL(ctx)
 		if err != nil {
-			return nil, hint, err
-		}
-		latestReq, err := http.NewRequestWithContext(ctx, http.MethodGet, latestURI.String(), nil)
-		if err != nil {
-			return nil, hint, err
-		}
-		latestRes, err := u.client.Do(latestReq)
-		if err != nil {
-			return nil, hint, err
-		}
-		defer latestRes.Body.Close()
-
-		err = httputil.CheckResponse(latestRes, http.StatusOK)
-		if err != nil {
-			return nil, hint, fmt.Errorf("unexpected response from archive_latest.txt: %w", err)
+			return nil, hint, fmt.Errorf("could not get compressed file URL: %w", err)
 		}
+		zlog.Debug(ctx).
+			Str("url", compressedURL.String()).
+			Msg("got compressed URL")
 
-		body, err := io.ReadAll(latestRes.Body) // Fine to use as expecting small number of bytes.
+		fp.requestTime, err = u.getLastModified(ctx, compressedURL)
 		if err != nil {
-			return nil, hint, err
+			return nil, hint, fmt.Errorf("could not get last-modified header: %w", err)
 		}
+	}
 
-		compressedFilename := string(body)
-		zlog.Debug(ctx).
-			Str("filename", compressedFilename).
-			Msg("requesting latest compressed file")
+	changed := map[string]bool{}
+	err = u.processChanges(ctx, cw, fp, changed)
+	if err != nil {
+		return nil, hint, err
+	}
 
-		uri, err := u.url.Parse(compressedFilename)
-		if err != nil {
-			return nil, hint, err
-		}
+	err = u.processDeletions(ctx, cw, fp, changed)
+	if err != nil {
+		return nil, hint, err
+	}
 
+	if processArchive {
```

Comment:
I'm kind of tempted to just make this an
```go
if compressedURL != nil {
```
But it feels like maybe it'd lose something in the readability.